cep18/src/utils.rs

```diff
@@ -136,13 +139,20 @@ pub fn get_named_arg_with_user_errors<T: FromBytes>(
             api_error::result_from(ret).map(|_| data)
         };
         // Assumed to be safe as `get_named_arg_size` checks the argument already
-        res.unwrap_or_revert_with(Cep18Error::FailedToGetArgBytes)
+        res.map_err(|err| {
+            let _ = runtime::emit_message(ERRORS, &format!("{err:?}").into());
```

Comment:
As discusses I believe that as soon as this PR hits feat-2.0 then it would be better for ceps to use that new method and remove the inner code of get_named_arg_with_user_errors to call this new method, and in this Core I would add native events messaging eventually.